ldap/servers/plugins/sync/sync_refresh.c

```diff
@@ -764,9 +765,15 @@ sync_send_deleted_entries(Slapi_PBlock *pb, Sync_UpdateNode *upd, int chg_count,
             } else {
                 /* max number of uuids to be sent in one sync info message */
                 syncUUIDs[uuid_index] = NULL;
-                sync_intermediate_msg(pb, LDAP_TAG_SYNC_ID_SET, cookie, (char **)&syncUUIDs);
+                for (size_t i = 0; i < uuid_index; i++) {
+                    ber_syncUUIDs[i] = (struct berval *) slapi_ch_malloc(sizeof(struct berval));
+                    ber_syncUUIDs[i]->bv_val = syncUUIDs[i];
+                    ber_syncUUIDs[i]->bv_len = 16;
+                }
+                sync_intermediate_msg(pb, LDAP_TAG_SYNC_ID_SET, cookie, ber_syncUUIDs);
                 for (size_t i = 0; i < uuid_index; i++) {
                     slapi_ch_free((void **)&syncUUIDs[i]);
+                    slapi_ch_free((void **)&ber_syncUUIDs[i]);
```

Comment:
I wonder if we should use `ber_bvfree()` here?
And add `ber_syncUUIDs[i] = NULL;` afterwards.